sway-lib-std/src/u128.sw

```diff
@@ -304,18 +304,18 @@ impl core::ops::Subtract for U128 {
 impl core::ops::Multiply for U128 {
     /// Multiply a `U128` with a `U128`. Panics of overflow.
     fn multiply(self, other: Self) -> Self {
-        let zero = U128::from((0, 0));
-        let one = U128::from((0, 1));
-
-        let mut x = self;
-        let mut y = other;
-        let mut result = U128::new();
-        while y != zero {
-            if (y & one).lower != 0 {
-                result += x;
-            }
-            x <<= 1;
-            y >>= 1;
+        // in case both of the `U128` upper parts are bigger than zero,
+        // it automatically means overflow, as any `U128` value
+        // is upper part multiplied by 2 ^ 64 + lower part
+        assert(!(self.upper != 0 && other.upper != 0));
```

Comment:
```suggestion
        assert(self.upper == 0 || other.upper == 0);
```
Waaay more readable. :)

Reply:
I'd leave as it is, as what I wanted to say is that I want to assure a situation in which both values don't have upper values, and not that I want at least one of them to zero. From the point of view of mathematic logic it's the same thing, but semantically, no.

Reply:
Semantically it's pretty much the same, really.  I looked at your original version for a minute just to work out what it was actually doing because of the double-negatives.

Reply:
Yeah I think Toby is right. Saying "We need at least one `upper` value to be 0 to avoid overflow" is pretty clear and requires less operators.